src/jit/codegenxarch.cpp

```diff
@@ -1367,6 +1367,15 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             {
                 assert(!isRegCandidate);
 
+#if defined(FEATURE_SIMD) && defined(_TARGET_X86_)
+                // Loading of TYP_SIMD12 (i.e. Vector3) variable
+                if (treeNode->TypeGet() == TYP_SIMD12)
+                {
+                    genLoadLclTypeSIMD12(treeNode);
+                    break;
+                }
+#endif // defined(FEATURE_SIMD) && defined(_TARGET_X86_)
```

Comment:
> Rather than special casing TYP_SIMD12 locals in GT_LCL_VAR/STORE_LCL_VAR and spill/reload code paths, another option to consider is to lower arg accesses as GT_IND(LCL_VAR_ADDR) or GT_STOREIND(LCL_VAR_ADDR) and thereby eliminating TYP_SIMD12 local accesses. Obviously this has the CQ downside that TYP_SIMD12 args are not enregistered and accessed from their stack location each time. Something to consider as a short term SIMD bring up on x86.

If I understand this suggestion correctly, I would convert call args as follows:
```
For call arg x (i.e., GT_CALL(x)):
1. if (x == GT_LCL_VAR)
  { bash x to GT_LCL_VAR_ADDR }
else
  { generate GT_STORE_LCL_VAR[tmp](x) to store it to a new lclvar temp
    set x = a new GT_LCL_VAR_ADDR[tmp] node
 }
2. convert call arg to GT_IND(GT_LCL_VAR_ADDR[tmp])
```
?

This might work to cause args to go through the existing AMD64 code path, but I don't see how this would handle spill/reload code paths. I guess the point is we wouldn't be spilling/reloading simd12 types, we would only be spilling/reloading simd16 locals, so we wouldn't need to change existing code that handles those already?